waku/v2/node/wakunode2.go

```diff
@@ -265,11 +266,9 @@ func New(opts ...WakuNodeOption) (*WakuNode, error) {
 		w.log.Error("creating peer connection strategy", zap.Error(err))
 	}
 
-	if w.opts.enableDiscV5 {
-		err := w.mountDiscV5()
-		if err != nil {
-			return nil, err
-		}
+	w.discoveryV5, err = w.opts.discv5Params.Service(w.opts.privKey, w.localNode, w.peerConnector, w.opts.prometheusReg, w.log, w.opts.discoveredNodes, w.opts.advertiseAddrs)
```

Comment:
This function is not very elegant due to the number of parameters it has (it's a code smell).